checker/jdk/nullness/src/javax/script/CompiledScript.java

```diff
@@ -0,0 +1,121 @@
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package javax.script;
+
+import java.util.Map;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * Extended by classes that store results of compilations.  State
+ * might be stored in the form of Java classes, Java class files or scripting
+ * language opcodes.  The script may be executed repeatedly
+ * without reparsing.
+ * <br><br>
+ * Each <code>CompiledScript</code> is associated with a <code>ScriptEngine</code> -- A call to an  <code>eval</code>
+ * method of the <code>CompiledScript</code> causes the execution of the script by the
+ * <code>ScriptEngine</code>.  Changes in the state of the <code>ScriptEngine</code> caused by execution
+ * of the <code>CompiledScript</code>  may visible during subsequent executions of scripts by the engine.
+ *
+ * @author Mike Grogan
+ * @since 1.6
+ */
+public abstract class CompiledScript {
+
+    /**
+     * Executes the program stored in this <code>CompiledScript</code> object.
+     *
+     * @param context A <code>ScriptContext</code> that is used in the same way as
+     * the <code>ScriptContext</code> passed to the <code>eval</code> methods of
+     * <code>ScriptEngine</code>.
+     *
+     * @return The value returned by the script execution, if any.  Should return <code>null</code>
+     * if no value is returned by the script execution.
+     *
+     * @throws ScriptException if an error occurs.
+     * @throws NullPointerException if context is null.
+     */
+
+    public abstract @Nullable Object eval(ScriptContext context) throws ScriptException;
+
+    /**
+     * Executes the program stored in the <code>CompiledScript</code> object using
+     * the supplied <code>Bindings</code> of attributes as the <code>ENGINE_SCOPE</code> of the
+     * associated <code>ScriptEngine</code> during script execution.  If bindings is null,
+     * then the effect of calling this method is same as that of eval(getEngine().getContext()).
+     * <p>.
+     * The <code>GLOBAL_SCOPE</code> <code>Bindings</code>, <code>Reader</code> and <code>Writer</code>
+     * associated with the default <code>ScriptContext</code> of the associated <code>ScriptEngine</code> are used.
+     *
+     * @param bindings The bindings of attributes used for the <code>ENGINE_SCOPE</code>.
+     *
+     * @return The return value from the script execution
+     *
+     * @throws ScriptException if an error occurs.
+     */
+    @SuppressWarnings("nullness") // as getEngine()'s implementation will never be returning null and same applies for getContext() method in ScriptEngine.java as specified in documentation. 
```

Comment:
What is the problem that you are claiming cannot happen?
If this is about the `ScriptContext ctxt` line, then put the `@SuppressWarnings` just before that line.  Make the scope as small as possible, not the whole method.